pkg/etcd/client.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package etcd
+
+import (
+	"context"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/ticdc/pkg/retry"
+	"go.etcd.io/etcd/clientv3"
+	"go.uber.org/zap"
+)
+
+// Client is a simple wrapper that adds retry to etcd RPC
+type Client struct {
+	cli *clientv3.Client
+}
+
+// Wrap warps a clientv3.Client that provides etcd APIs required by TiCDC.
+func Wrap(cli *clientv3.Client) *Client {
+	return &Client{cli: cli}
+}
+
+// Unwrap returns a clientv3.Client
+func (c *Client) Unwrap() *clientv3.Client {
+	return c.cli
+}
+
+func retryRPC(rpcName string, etcdRPC func() error) error {
+	// By default, PD etcd sets [3s, 6s) for election timeout.
+	// Some rpc could fail due to etcd errors, like "proposal dropped".
+	// 6.59s = \sum_{n=0}^{4} 0.5*1.5^n
+	return retry.Run(500*time.Millisecond, 4+1, // +1 for the inital request.
```

Comment:
Better to increase this timeout to  2 * max_election_timeout, to handle the case that two PDs restarted. (the first election maybe failed)